vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/ItemClickListenerIT.java

```diff
@@ -79,6 +79,32 @@ public void doubleClickCheckboxInCell_noEventsFired() {
         checkbox.doubleClick();
         Assert.assertEquals("", getClickMessage());
         Assert.assertEquals("", getDoubleClickMessage());
+        Assert.assertEquals("", getColumnClickMessage());
+    }
+
+    @Test
+    public void clickCell_columnNameAvailable() {
+        grid.getCell(0, 0).click();
+        Assert.assertEquals("Name", getColumnClickMessage());
+    }
+
+    @Test
+    public void clickComponentInCell_columnNameAvailable() {
+        grid.getCell(0, 0).click();
+        TestBenchElement checkbox = grid.getCell(0, 1).$("vaadin-checkbox")
+                .first();
+        checkbox.click();
+        Assert.assertEquals("Name", getColumnClickMessage());
+    }
+
+    @Test
+    public void doubleClickCell_columnNameAvailable() {
+        grid.getCell(0, 0).doubleClick();
+        Assert.assertEquals("Name", getColumnClickMessage());
```

Comment:
This is kind of ugly, because it depends on the click-listener adding the "Name"-text twice before the double-click event is fired, which sets the value to single "Name". Would be cleaner to have separate messages for click and dblclick.

Reply:
It could turn into a false-positive in many ways, e.g.:
1. For some reason, `dblclick` gets fired before the second `click` on some browser.
2. Someone is not careful and changes the test-page so that `click` also calls `setText("name")` instead of adding text.

Reply:
Agree. Didn't fully understand how tests work here, my fault. Fixed.